### PR TITLE
Added e2e testcases

### PR DESCRIPTION
**Pull request by SjoerdHilhorst**
to locally use e2e test you can run:
gulp protractor
gulp protractor test/e2e/protractor.conf.js --suite <provider, requester, sponsor>

### CODE REVIEW

**Review by maxvisser — CHANGES_REQUESTED** [review]
works; sample file should be moved to right directory and aswell be cleaned up a bit. Alot of unnessecary lines. As well as some variables are missing.

Could you also check why this test failed:
1) testing basic requester functionality: opens and closes start activation modal
  Message:
    Failed: No element found using locator: By(css selector, [ng-click="$ctrl.openAuthCodePopup()"])

as you suggested in chat, it could be that the test was still logged in.

File: src/sources/forus-platform/pug/tpl/pages/sign-up.pug [hunk 1]
```diff
@@ -212,7 +212,7 @@
 
             .step-footer
                 a(translate="sign_up.buttons.cancel" ng-click="$ctrl.back()" ng-if="!$ctrl.signedIn").button.button-default
-                a(translate="sign_up.buttons.next" ng-click="$ctrl.next()").button.button-primary
+                a(translate="sign_up.buttons.next" ng-click="$ctrl.next()" id='next-4').button.button-primary
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/pages/sign-up.pug]: I would use " instead of '

**Comment by maxvisser** [comment]
single format for id; suggestion is Snake Case example:
signup_next

so not :
3
next-3

element(by.css('[ui-sref="offices({organization_id: $root.activeOrganization.id})"]')).click()

replace by ID's

**Comment by maxvisser** [comment]
https://github.com/teamforus/forus-frontend/pull/1047/files/07399a24ab8f1e3b8713d8896ebfcf137b4a5255#diff-ff9128d90e5fb486bb30f64d2637b130 is not fixed yet

**Comment by maxvisser** [comment]
As you use classes, ui-refs and or other existing html tags your test now can easily break by changing styles or texts. I would suggest to still change all by.css, by.className, by.model and others to by.id

Probably this means you need to go through alot of pug files. If you need help with this please let me know.

File: src/sources/forus-platform/pug/tpl/pages/products.pug [hunk 2]
```diff
@@ -4,7 +4,7 @@
         //- information-block(text="'products.cannot_delete'" type="'danger'")
         .card-heading
             | {{ "products.offers" | translate }}
-            a(ng-click="$ctrl.addProduct()").link
+            a(ng-click="$ctrl.addProduct()" id="addProduct").link
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/pages/products.pug]: Snakecase

File: src/sources/forus-platform/pug/tpl/pages/landing/sign-up.pug [hunk 3]
```diff
@@ -149,7 +149,7 @@
 
                 .card-section.card-section-primary.text-center
                     a(ng-click="$ctrl.skipToStep(5)" translate="sign_up.buttons.skip").button.button-default
-                    button(ng-click="$ctrl.next()" translate="sign_up.buttons.next").button.button-primary
+                    button(ng-click="$ctrl.next()" id='3' translate="sign_up.buttons.next").button.button-primary
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/pages/landing/sign-up.pug]: Snakecase and '' -> to ""

**Comment by SjoerdHilhorst** [comment]
https://www.protractortest.org/#/style-guide
> Prefer protractor-specific locators such as by.model and by.binding.

protractor zegt dat by.model wel een goede locator is, maar jouw keuze natuurlijk

**Comment by maxvisser** [comment]
I would just add each last part of the ng-model as an id field
example:
```html
<input
ng-model="$ctrl.signUpForm.values.records.primary_email"
id="primary_email" />
```
if any of these variables will change $ctrl.signUpForm.values.records your test would fail. With using the ID's I can't really see how it could fail. (only if the field is removed or altered.)

File: src/package-lock.json [hunk 4]
```diff
@@ -7485,21 +7940,26 @@
       "version": "3.0.0",
       "resolved": "https://registry.npmjs.org/p-map/-/p-map-3.0.0.tgz",
       "integrity": "sha512-d3qXVTF/s+W+CdJ5A29wywV2n8CQQYahlgz2bFiA+4eVNJbHJodPZ+/gXwPGh0bOqA+j8S+6+ckmvLGPk1QpxQ==",
-      "dev": true,
       "requires": {
         "aggregate-error": "^3.0.0"
       }
     },
     "pako": {
+<<<<<<< HEAD
```
Comment (maxvisser) [inline src/package-lock.json]: @SjoerdHilhorst regenerate lock file

File: src/sources/forus-platform/pug/tpl/pages/landing/sign-up.pug (hunk 3 above)
Comment (maxvisser) [reply src/sources/forus-platform/pug/tpl/pages/landing/sign-up.pug]: forgot this one?

**Review by maxvisser — CHANGES_REQUESTED** [review]
Couuld you still fix the comments I had left

**Comment by maxvisser** [comment]
revert package-lock.json as there were merge conflicts pushed

@dev-rminds could you code review this?

**Comment by SjoerdHilhorst** [comment]
for some weird reason some git conflicts did not show up in vscode, but i think it's all fixed. Updated the dashboard testcases for the new way of selection a fund. dashboard testcases are (lightly) dependent on LoremDbSeeder so you need to run that first and then run the tests

**Comment by GerbenBosschieter** [comment]
@maxvisser What about this one. It's already open for a while. What is the hold up ?

**Comment by maxvisser** [comment]
@dev-rminds will review it after finishing notification/digest. He need some time to get into this issue as the scope is pretty big

**Comment by jamalv** [comment]
@maxvisser could you give rough estimations on both those things? I think since merging this is blocked by the check of valentin, that it's pretty important to review this. Since I agree with gerben that this has been open for quite a while now.

**Comment by maxvisser** [comment]
I think this weekend there should be time. If the other tasks are finished.

**Comment by maxvisser** [comment]
fine to merge!

Questions: Is there a reason that:
local version is not checking the email
remote version is not checking the validator dashboard

- added some env variables like email port etc.

- added a test task (which combines browserstack and protractor)

feedback:
- some functions, are 'one time use'-functions, not sure if this is a best practice. For now its fine

some test were failing 
- in a random order each run
image

- we got a ssl routine, wrong version number error

Tips:
protractor can be used without gulp (@dev-rminds will check this out, if output style is nicer)
@dev-rminds would create everything within the tests (instead of using a seeder)
later phase create cli client -> would help with directly testing the api
Later:
- add more scenario's

**Review by maxvisser — APPROVED** [review]

**Comment by SjoerdHilhorst** [comment]
What exactly do you mean with local and remote?
and are these also questions or statements? 
> added some env variables like email port etc.

>added a test task (which combines browserstack and protractor

The mail listener was still something left over from when i tested complete flows. and currently not used for testing. I could remove it, but i can probably also make some useful testcases with it.

Could you clarify on not checking validator dashboard, when running it should just execute all test suites.

protractor is already able to run without gulp, run: `protractor test/e2e/protractor.conf.js`

**Pull request merged**